cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/io/cdap/cdap/etl/api/connector/ConnectorSpec.java

```diff
@@ -31,10 +31,22 @@ public class ConnectorSpec {
   // schema is null when the connector is unable to retrieve it from the resource
   private final Schema schema;
   private final Set<PluginSpec> relatedPlugins;
+  private final Set<String> supportedSampleTypes;
 
-  private ConnectorSpec(@Nullable Schema schema, Set<PluginSpec> relatedPlugins) {
+  private ConnectorSpec(@Nullable Schema schema,
+                        Set<PluginSpec> relatedPlugins,
+                        Set<String> supportedSampleTypes) {
     this.schema = schema;
     this.relatedPlugins = relatedPlugins;
+    if (supportedSampleTypes != null) {
```

Comment:
Any specific reason to handle it as nullable? It's a private constructor, so it's easier to make all callers supply a value

Reply:
I believe the intention was to account for connectors without supported sample types, but all connectors use the Builder which supplies an empty Set anyways, so it is likely unnecessary. I have removed it.